code/datums/datumvars.dm

```diff
@@ -1,220 +1,103 @@
-					for ( var j = 0; j < lis_new.length; ++j )
-					{
-						var li1 = lis\[j\];
-						if (j == 0){
-							li1.style.backgroundColor = "#ffee88";
-						}else{
-							li1.style.backgroundColor = "white";
-						}
-					}
-				}
-
-
-
-				function selectTextField(){
-					var filter_text = document.getElementById('filter');
-					filter_text.focus();
-					filter_text.select();
-
-				}
-
-				function loadPage(list) {
-
-					if(list.options\[list.selectedIndex\].value == ""){
-						return;
-					}
-
-					location.href=list.options\[list.selectedIndex\].value;
-
-				}
-			</script> "}
-
-	body += "<body onload='selectTextField(); updateSearch()' onkeyup='updateSearch()'>"
-
-	body += "<div align='center'><table width='100%'><tr><td width='50%'>"
+	title = "[D] (\ref[D]) = [type]"
 
+	var/sprite_text
 	if(sprite)
-		body += "<table align='center' width='100%'><tr><td><img src='view_vars_sprite.png'></td><td>"
-	else
-		body += "<table align='center' width='100%'><tr><td>"
-
-	body += "<div align='center'>"
+		sprite_text = "<img src='view_vars_sprite.png'></td><td>"
+	var/list/atomsnowflake = list()
 
 	if(istype(D,/atom))
 		var/atom/A = D
 		if(isliving(A))
-			body += "<a href='?_src_=vars;rename=\ref[D]'><b>[D]</b></a>"
+			atomsnowflake += "<a href='?_src_=vars;rename=[refid]'><b>[D]</b></a>"
 			if(A.dir)
-				body += "<br><font size='1'><a href='?_src_=vars;rotatedatum=\ref[D];rotatedir=left'><<</a> <a href='?_src_=vars;datumedit=\ref[D];varnameedit=dir'>[dir2text(A.dir)]</a> <a href='?_src_=vars;rotatedatum=\ref[D];rotatedir=right'>>></a></font>"
+				atomsnowflake += "<br><font size='1'><a href='?_src_=vars;rotatedatum=[refid];rotatedir=left'><<</a> <a href='?_src_=vars;datumedit=[refid];varnameedit=dir'>[dir2text(A.dir)]</a> <a href='?_src_=vars;rotatedatum=[refid];rotatedir=right'>>></a></font>"
 			var/mob/living/M = A
-			body += "<br><font size='1'><a href='?_src_=vars;datumedit=\ref[D];varnameedit=ckey'>[M.ckey ? M.ckey : "No ckey"]</a> / <a href='?_src_=vars;datumedit=\ref[D];varnameedit=real_name'>[M.real_name ? M.real_name : "No real name"]</a></font>"
-			body += {"
-			<br><font size='1'>
-			BRUTE:<font size='1'><a href='?_src_=vars;mobToDamage=\ref[D];adjustDamage=brute'>[M.getBruteLoss()]</a>
-			FIRE:<font size='1'><a href='?_src_=vars;mobToDamage=\ref[D];adjustDamage=fire'>[M.getFireLoss()]</a>
-			TOXIN:<font size='1'><a href='?_src_=vars;mobToDamage=\ref[D];adjustDamage=toxin'>[M.getToxLoss()]</a>
-			OXY:<font size='1'><a href='?_src_=vars;mobToDamage=\ref[D];adjustDamage=oxygen'>[M.getOxyLoss()]</a>
-			CLONE:<font size='1'><a href='?_src_=vars;mobToDamage=\ref[D];adjustDamage=clone'>[M.getCloneLoss()]</a>
-			BRAIN:<font size='1'><a href='?_src_=vars;mobToDamage=\ref[D];adjustDamage=brain'>[M.getBrainLoss()]</a>
-			STAMINA:<font size='1'><a href='?_src_=vars;mobToDamage=\ref[D];adjustDamage=stamina'>[M.getStaminaLoss()]</a>
-			</font>
-
-
+			atomsnowflake += {"

```

Comment:
Could be per type so we can have fancy displays for common objects, then again, don't expect anyone to waste time on that so probably pointless.